### PR TITLE
Add new controls in all_controls benchmark 

### PR DESCRIPTION
**Pull request by khushboo9024**
### Checklist

1. [x] cluster_role_binding_default_service_account_binding_not_active
2. [x] cronjob_container_argument_request_timeout_appropriate
3. [x] cronjob_container_secrets_defined_as_files
4. [x] daemonset_container_host_port_not_specified
5. [x] daemonset_container_argument_request_timeout_appropriate
6. [x] daemonset_container_secrets_defined_as_files
7. [x] deployment_container_host_port_not_specified
8. [x] deployment_container_argument_request_timeout_appropriate
9. [x] deployment_container_secrets_defined_as_files
10. [x] ingress_nginx_annotations_snippets_alias_not_used
11. [x] ingress_nginx_annotations_all_snippets_not_used
12. [x] ingress_nginx_annotations_snippets_lua_code_not_used
13. [x] job_container_host_port_not_specified
14. [x] job_container_argument_request_timeout_appropriate
15. [x] job_container_secrets_defined_as_files
16. [x] pod_container_host_port_not_specified
17. [x] pod_service_account_token_enabled
18. [x] pod_container_secrets_defined_as_files
19. [x] pod_container_run_as_user_10000
20. [x] pod_container_argument_request_timeout_appropriate
21. [x] pod_template_container_host_port_not_specified
22. [x] pod_template_container_argument_request_timeout_appropriate
23. [x] pod_template_container_secrets_defined_as_files
24. [x] replicaset_container_host_port_not_specified
25. [x] replicaset_container_argument_request_timeout_appropriate
26. [x] replicaset_container_secrets_defined_as_files
27. [x] replication_controller_container_host_port_not_specified
28. [x] replication_controller_container_argument_request_timeout_appropriate
29. [x] replication_controller_container_secrets_defined_as_files
30. [x] role_with_rbac_escalate_permissions
31. [x] role_with_bind_cluster_role_bindings
32. [x] cluster_role_with_validating_or_mutating_admission_webhook_configurations
33. [x] role_with_rbac_approve_certificate_signing_requests
34. [x] role_binding_default_service_account_binding_not_active
35. [x] service_no_tiller_service
36. [x] service_no_tiller_deployed
37. [x] statefulset_container_host_port_not_specified
38. [x] statefulset_container_argument_request_timeout_appropriate
39. [x] statefulset_container_secrets_defined_as_files

### CODE REVIEW

File: query/role.sp [hunk 1]
```diff
@@ -56,3 +56,229 @@ query "role_with_wildcards_used" {
       _ctx;
   EOQ
 }
+
+query "role_with_rbac_escalate_permissions" {
+  sql = <<-EOQ
+    with role_with_escalate as (
+      select
+        uid,
+         count(*) as num
```
Comment (madhushreeray30) [inline query/role.sp:65]:
```suggestion
        count(*) as num
```

File: query/role.sp [hunk 2]
```diff
@@ -56,3 +56,229 @@ query "role_with_wildcards_used" {
...
+          rule -> 'resources' @> '["roles"]'
+          or rule -> 'resources' @> '["clusterroles"]'
+        )
+        and rule -> 'verbs'  @> '["escalate"]'
+      group by
+        uid
+      union
+      select
+        uid,
+         count(*) as num
```
Comment (madhushreeray30) [inline query/role.sp:81]:
```suggestion
        uid,
        count(*) as num
```

File: query/role.sp [hunk 3]
```diff
@@ -56,3 +56,229 @@ query "role_with_wildcards_used" {
...
+      from
+        kubernetes_role,
+        jsonb_array_elements(rules) rule
+      where
+        rule -> 'apiGroups' @> '["rbac.authorization.k8s.io"]'
+        and (
+          rule -> 'resources' @> '["rolebindings"]'
+          or rule -> 'resources' @> '["clusterrolebindings"]'
+        )
+        and rule -> 'verbs'  @> '["bind"]'
```
Comment (madhushreeray30) [inline query/role.sp:167]:
```suggestion
        and rule -> 'verbs' @> '["bind"]'
```

File: query/service.sp [hunk 4]
```diff
@@ -36,3 +36,51 @@ query "service_type_forbidden" {
...
+    select
+      coalesce(uid, concat(path, ':', start_line)) as resource,
+      name,
+      case
+        when labels ->> 'app' = 'helm' or labels ->> 'name' = 'tiller' then 'alarm'
+        else 'ok'
+      end as status,
+      case
+        when labels ->> 'app' = 'helm' or labels ->> 'name' = 'tiller' then name || ' has tiller deployed.'
+        else  name || ' tiller not deployed.'
```
Comment (madhushreeray30) [inline query/service.sp:79]:
```suggestion
        else  name || ' does not have tiller deployed.'
```

File: query/statefulset.sp [hunk 5]
```diff
@@ -2491,3 +2491,80 @@ query "statefulset_container_strong_kube_apiserver_cryptographic_ciphers" {
...
+        when l.container_name is not null and (p.value -> 'command') @> '["kube-apiserver"]' and ((l.value) ~ '^(\d{1,2}[h])(\d{1,2}[m])?(\d{1,2}[s])?$|^(\d{1,2}[m])?(\d{1,2}[s])?$|^(\d{1,2}[s])$') then p.value ->> 'name' || ' request-timeout set appropriate.'
+        else p.value ->> 'name' || ' request-timeout set inappropriate.'
+      end as reason,
+      p.statefulset_name as statefulset_name
+      ${local.tag_dimensions_sql}
+      ${local.common_dimensions_sql}
+    from
+      container_name_with_statefulset_name as p
+      left join container_list as l
+        on p.value ->> 'name' = l.container_name and p.statefulset_name = l.statefulset
```
Comment (madhushreeray30) [inline query/statefulset.sp:2568]:
```suggestion
      left join container_list as l on p.value ->> 'name' = l.container_name and p.statefulset_name = l.statefulset;
```

**Review by madhushreeray30 — COMMENTED** [review]
Please take a look at the comments, thanks!

File: query/ingress.sp [hunk 6]
```diff
@@ -16,4 +16,64 @@ query "ingress_default_namespace_used" {
...
+query "ingress_nginx_annotations_snippets_lua_code_not_used" {
+  sql = <<-EOQ
+    select
+      coalesce(uid, concat(path, ':', start_line)) as resource,
+      case when a.key like '%snippet%' and a.value ~ '(lua_|_lua|_lua_|kubernetes\.io)' then 'alarm'
+        else 'ok'
+      end as status,
+      case
+        when a.key like '%snippet%' and a.value ~ '(lua_|_lua|_lua_|kubernetes\.io)' then a.key || ' annotation snippet contains lua code execution.'
+        else a.key || ' annotation snippet not containing lua code execution.'
```
Comment (misraved) [inline query/ingress.sp:70]:
```suggestion
        else a.key || ' annotation snippet does not contain lua code execution.'
```

File: query/role.sp [hunk 7]
```diff
@@ -56,3 +56,229 @@ query "role_with_wildcards_used" {
...
+    )
+    select
+      coalesce(r.uid, concat(r.path, ':', r.start_line)) as resource,
+      case
+        when e.num > 0 then 'alarm'
+        else 'ok'
+      end as status,
+      case
+        when e.num > 0  then name || ' contains ' || e.num || ' RBAC cluster role validating or mutating admission webhook configurations permissions.'
+        else name || ' does not contains any bind role bindings or cluster role validating or mutating admission webhook configurations permissions.'
```
Comment (misraved) [inline query/role.sp:239]:
```suggestion
        else name || ' does not contain any bind role bindings or cluster role validating or mutating admission webhook configurations permissions.'
```

**Review by misraved — APPROVED** [review]

**Pull request merged**